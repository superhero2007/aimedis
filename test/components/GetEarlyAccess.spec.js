import React from 'react';
import { mount } from 'enzyme';

import GetEarlyAccess from 'components/GetEarlyAccess';

function setup() {
  return mount(<GetEarlyAccess />);
}

describe('GetEarlyAccess', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();

    const input = wrapper.find('.app__get__early__access__input');
    input.simulate('blur');
    expect(wrapper.state().flag).toEqual('INVALIDATE');

    input.simulate('blur', { target: { value: 'email@outlook.com' } });
    expect(wrapper.state().flag).toEqual('VALIDATE');
  });
});
