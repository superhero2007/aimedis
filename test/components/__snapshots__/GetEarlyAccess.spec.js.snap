// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetEarlyAccess should render properly 1`] = `"<div class=\\"app__get__early__access\\"><input class=\\"app__get__early__access__input\\" type=\\"text\\" placeholder=\\"Your Email Address\\"><button type=\\"button\\" class=\\"app__get__early__access__button btn btn__primary\\">Get <span class=\\"d-none d-lg-inline\\">Early </span>Access</button></div>"`;
